src/main/scala/dotty/dokka/ScalaPageCreator.scala

```diff
@@ -0,0 +1,91 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.base.translators.documentables.{DefaultPageCreator, PageContentBuilder, PageContentBuilder$DocumentableContentBuilder}
+import org.jetbrains.dokka.base.signatures.SignatureProvider
+import org.jetbrains.dokka.base.transformers.pages.comments.CommentsToContentConverter
+import org.jetbrains.dokka.transformers.documentation.DocumentableToPageTranslator
+import org.jetbrains.dokka.utilities.DokkaLogger
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.pages._
+import collection.JavaConverters._
+import org.jetbrains.dokka.model.properties._
+
+
+class ScalaPageCreator(
+    commentsToContentConverter: CommentsToContentConverter,
+    signatureProvider: SignatureProvider,
+    val logger: DokkaLogger
+) extends DefaultPageCreator(commentsToContentConverter, signatureProvider, logger) {
+    override def pageForClasslike(c: DClasslike): ClasslikePageNode = {
+        val res = super.pageForClasslike(c)
+        if(c.isInstanceOf[DClass]){
+            val clazz = c.asInstanceOf[DClass]
+            clazz.get(ClasslikeExtension).companion.map( co =>
+                res.modified(
+                    res.getName,
+                    res.getContent,
+                    res.getDri,
+                    res.getEmbeddedResources,
+                    (res.getChildren.asScala ++ List(
+                        ClasslikePageNode(
+                            co.getName + "$",
+                            contentForClasslike(co),
+                            Set(co.getDri).asJava,
+                            co,
+                            (co.getClasslikes.asScala.map(pageForClasslike(_)) ++
+                            co.getFunctions.asScala.map(pageForFunction(_))).asJava,
+                            Nil.asJava
+                        )
+                    )).asJava
+                )
+            ).getOrElse(res)
+        } else res
+    }
+
+    override def contentForClasslike(c: DClasslike): ContentGroup = {
+        val defContent = super.contentForClasslike(c)
+        if(c.isInstanceOf[DClass]){
+            val clazz = c.asInstanceOf[DClass]
+            clazz.get(ClasslikeExtension).companion.map(co => {
+                val addedContent = PageContentBuilder(commentsToContentConverter, signatureProvider, logger).contentFor(
+                    c,
+                    ContentKind.Main,
+                    Set().asJava,
+                    PropertyContainer.Companion.empty(),
+                    c.getSourceSets, builder => {
```

Comment:
We can pull this lambda to inner method so it is more readble

Reply:
Done